File: Database System Management/ModelComponent/Model/ModelEvent.cs
using System.Collections.Generic;
using ModelComponent.Domain;
using ModelComponent.Entity;
using ModelComponent.Repository;

namespace ModelComponent.Model
{
    public class ModelEvent : Model<Event, EventEntity>
    {
        private readonly IRepositoryEntity<Player, PlayerEntity> _repositoryPlayer;
        private readonly IRepositoryRelation<Player, PlayerEntity, Event, EventEntity, PlayerEventEntity>
            _repositoryRelation;

        public ModelEvent(string database) : base(database)
        {
            Repository = RepositoryFactory.GetInstance<Event, EventEntity>(PreferredDatabaseSource, RepositoryType.Event);
            _repositoryPlayer = RepositoryFactory.GetInstance<Player, PlayerEntity>(PreferredDatabaseSource,
                RepositoryType.Player);
            _repositoryRelation =
                RepositoryFactory.GetRelation<Player, PlayerEntity, Event, EventEntity, PlayerEventEntity>(
                    PreferredDatabaseSource, RepositoryType.PlayerEvent);
        }

        public ModelEvent()
        {
            Repository = RepositoryFactory.GetInstance<Event, EventEntity>(PreferredDatabaseSource, RepositoryType.Event);
            _repositoryPlayer = RepositoryFactory.GetInstance<Player, PlayerEntity>(PreferredDatabaseSource,
                RepositoryType.Player);
            _repositoryRelation =
                RepositoryFactory.GetRelation<Player, PlayerEntity, Event, EventEntity, PlayerEventEntity>(
                    PreferredDatabaseSource, RepositoryType.PlayerEvent);
        }

        public void AddEventToPlayer(int idEvent, int idPlayer)
        {
            var item = _repositoryPlayer.FindElementById(idPlayer);
            var element = Repository.FindElementById(idEvent);
            if (item != null && element != null)
            {
                _repositoryRelation.Add(item, element);
            }
        }

        public void DeleteEventFromPlayer(int idEvent, int idPlayer)
        {
            var item = _repositoryPlayer.FindElementById(idPlayer);
            var element = Repository.FindElementById(idEvent);
            if (item != null && element != null)
            {
                _repositoryRelation.Delete(item, element);
            }
        }

        public List<Player> GetAllEventPlayers(int idEvent)
        {
            var element = Repository.FindElementById(idEvent);
            return element != null ? _repositoryRelation.GetAllT(element) : new List<Player>();
        }
    }
}

File: Advanced Programming Methods/Advanced Programming Methods Server-Client/mpp.model/src/main/java/model/ModelEvent.java
package model;

import configuration.*;
import domain.Event;
import domain.Player;
import error.Errors;
import repository.RepositoryEntityProtocol;
import repository.RepositoryFactory;
import repository.RepositoryRelationshipDatabase;
import repository.RepositoryType;

import java.util.ArrayList;

/**
 * @author Alexandru Stoica
 * @version 1.0
 */

public class ModelEvent extends Model<Event, Integer> {

    private RepositoryRelationshipDatabase repositoryRelationship;
    private RepositoryEntityProtocol repositoryPlayer;

    public ModelEvent(String database) {
        this.preferredDatabaseSource = database;
        this.checkDatabase();

        ConfigurationProtocol configurationPlayer =
                ConfigurationFactory.build(ConfigurationType.PLAYER);
        ConfigurationRelationProtocol configurationRelation =
                new PlayerEventConfiguration();
        ConfigurationProtocol configurationEvent =
                ConfigurationFactory.build(ConfigurationType.EVENT);

        this.repository = RepositoryFactory.getRepository(
                RepositoryType.EVENT,
                preferredDatabaseSource,
                configurationEvent
        );
        this.repositoryRelationship = new RepositoryRelationshipDatabase<Player, Event, Integer, Integer>(
                preferredDatabaseSource,
                configurationRelation
        );
        this.repositoryPlayer =
                RepositoryFactory.getRepository(
                        RepositoryType.PLAYER,
                        preferredDatabaseSource,
                        configurationPlayer
                );
    }

    public ModelEvent() {
        this(database);
    }


    @SuppressWarnings("unchecked")
    public void addPlayerToEvent(Integer IDPlayer, Integer IDEvent) {
        try {
            Player player = (Player) repositoryPlayer.findElementById(IDPlayer);
            Event event = (Event) repository.findElementById(IDEvent);
            if (event != null && player != null) {
                repositoryRelationship.add(player, event);
            }
        } catch (Errors errors) {
            handleErrors(errors);
        }
    }

    @SuppressWarnings("unchecked")
    public void deletePlayerFromEvent(Integer IDPlayer, Integer IDEvent) {
        try {
            Player player = (Player) repositoryPlayer.findElementById(IDPlayer);
            Event event = (Event) repository.findElementById(IDEvent);
            if (event != null && player != null) {
                repositoryRelationship.delete(player, event);
            }
        } catch (Errors errors) {
            handleErrors(errors);
        }
    }

    @SuppressWarnings("unchecked")
    public ArrayList<Player> getPlayersFromEvent(Integer IDEvent) {
        try {
            return repositoryRelationship.getTObjectsByID(IDEvent);
        } catch (Errors errors) {
            handleErrors(errors);
            return new ArrayList<>();
        }
    }

}